lib/index.js

```diff
@@ -196,35 +168,27 @@ function handleTemplateCache (options, str) {
 exports.compile = function(str, options){
   var options = options || {}
     , filename = options.filename
-      ? utils.stringify(options.filename)
+      ? stringify(options.filename)
       : 'undefined'
     , fn;
 
   str = String(str);
 
-  var parsed = parse(str, options);
+  var parsed = compileBody(str, options);
   if (options.compileDebug !== false) {
     fn = [
-        'var jade_debug = [ new jade.DebugItem( 1, ' + filename + ' ) ];'
+        'var jade_debug_filename, jade_debug_line;'
       , 'try {'
       , parsed.body
       , '} catch (err) {'
-      , '  jade.rethrow(err, jade_debug[0].filename, jade_debug[0].lineno' + (options.compileDebug === true ? ',' + utils.stringify(str) : '') + ');'
+      , '  jade.rethrow(err, jade_debug_filename, jade_debug_line' + (options.compileDebug === true ? ', jade_debug_sources[jade_debug_filename]' : '') + ');'
       , '}'
```

Comment:
this is much better - it always bugged me why we store this big array only to use its first member